burn-train/src/renderer/tui/metric_numeric.rs

```diff
@@ -107,6 +107,10 @@ impl NumericMetricsState {
     /// Handle the current event.
     pub(crate) fn on_event(&mut self, event: &Event) {
         if let Event::Key(key) = event {
+            match key.kind {
+                KeyEventKind::Press | KeyEventKind::Repeat => (),
+                KeyEventKind::Release => return,
+            }
```

Comment:
I think the opposite would be preferred, we should perform the action on key release. But thanks for the fix :)

Reply:
I tested both ways on Windows and I prefer the behavior when the Press and Repeat events are enabled as @amfaber proposed, feels more natural and the terminal stays responsive when holding down the keys, at least on my system configuration. Would be great to have additional feedback on this.

Reply:
👍
That is probably more "idiomatic"

Reply:
> 👍 That is probably more "idiomatic"

Oops, replied without refreshing my browser

From general gui experience, I greatly prefer actions to happen on release if they are "cancelable" with something like esc or by performing another action, such that one can change ones mind at the last moment. For something as simple as this that is instantly undoable (by pressing the opposing button) and that isn't cancelable, I have no strong feelings for either Press or Release

Reply:
On Press has the advantage of being instantaneous as soon as the key is pressed, it can also feels more snappy. With On Release nothing happens while the key is pressed which may not be ideal.

I guess we have three possible ergonomics here:
- 1) original version: Press and Repeat On / Release Off
- 2) opposite version: Press and Repeat Off / Release On
- 3) alternate version: Press On / Release and Repeat Off

 I like better 2) and 3) with Press On. But I have also no strong feeling about it so whatever works if fine.

Reply:
A quick test showed that holding down the button actually sends repeated "Press" events, not "Repeated" (on my windows 11 atleast). Using "Release" also felt more sluggish, as  @syl20bnr pointed out. With this in mind I suggest we keep the original setup of 
```rust
match key.kind {
    KeyEventKind::Press | KeyEventKind::Repeat => (),
    KeyEventKind::Release => return,
}
```

Reply:
I tested on MacOS with iTerm (which is widely used) and `Release` does not work. Only `Press` works. Furthermore `Repeat` does not appear to fire [1] and `Press` acts also as a `Repeat` which means that holding an arrow key down cycles through metrics. Unfortunately iTerm is quite slow and the experience is not as smooth as on Windows, keeping the right arrow key pressed select a random metric every second or so.

[1] tried with a different arms order like:

```rust
            match key.kind {
                KeyEventKind::Release | KeyEventKind::Repeat => return,
                KeyEventKind::Press => (),
            }
```

Reply:
It's a bit more snappy in Alacritty on MacOS but not so much, might be my old Intel Mac that is slow 😝